X3 doesn't compile on g++ 4.8 cxxstd=11

https://travis-ci.org/boostorg/boost/jobs/308195225

```
./boost/spirit/home/x3/support/context.hpp:28:18: error: expected primary-expression before ‘auto’
         decltype(auto) get(ID_ id) const
                  ^
```

These tests should probably be guarded by `requires`.

Also,

```
In file included from libs/spirit/test/x3/x3_variant.cpp:9:
./boost/spirit/home/x3/support/ast/variant.hpp:134:50: error: no template named 'remove_reference_t' in namespace 'std'; did you mean simply 'remove_reference_t'?
                                                ,std::remove_reference_t<T>
```

Looks like X3 wants C++14.

--- a/test/x3/Jamfile
+++ b/test/x3/Jamfile
@@ -8,6 +8,7 @@
 #   http://www.boost.org/LICENSE_1_0.txt)
 #==============================================================================
 
+import config : requires ;
 import modules ;
 import feature ;
 import testing ;
@@ -18,9 +19,20 @@ project spirit-x3
     : requirements
         <include>.
         <c++-template-depth>512
-        <toolset>gcc:<cxxflags>-std=c++1y
-        <toolset>clang:<cxxflags>-std=c++1y
-        <toolset>darwin:<cxxflags>-std=c++1y
+        [ requires
+            # Assume all the cxx11 checks succeed if any of cxx14 does.
+            #cxx14_binary_literals               # grep -Er "[0-9]+b[0-9]+" *
+            #cxx14_constexpr
+            cxx14_decltype_auto                 # grep -r "decltype(auto)" *
+            #cxx14_digit_separators              # grep -Er "[0-9]+'[0-9]+" *
+            cxx14_generic_lambdas               # grep -Er "]\s*\\([^\\)]*auto" *
+            #cxx14_hdr_shared_mutex              # grep -r "shared_mutex" *
+            #cxx14_initialized_lambda_captures   # grep -Er "\\[[^=\\]]+=" *
+            #cxx14_aggregate_nsdmi
+            cxx14_return_type_deduction         # grep -Er "auto[^\\(=\\)]+\(" *
+            #cxx14_std_exchange                  # grep -r "exchange" *
+            #cxx14_variable_templates
+        ]
     ;
 
 ###############################################################################

